# 局域作用域坑

# count = 1
# def func():
#     count +=1  # 因为局部作用域没有定义count 所以会报错！！
#                # 对于全局作用域中的变量，只能引用，不能修改！
#     print(count)
#
# func()


# 在函数中，如果你定义了一个变量，但是在定义这个变量之前对其引用了，那么解释器认为：语法存在问题
# 你应该在使用之前先定义
count = 1
def func():
    print(count)
    #count = 3 加了这句，就有问题了。因为count 在全局作用域中
    # 并且，对于变量的引用，应该是先定义，后引用，而不是先引用，后定义。这个顺序也会让解释器产生歧义，count究竟是局部变量还是全局变量
func()




